src/Schema/Domain/Content/NameHelper.php

```diff
@@ -18,9 +19,21 @@ class NameHelper
      */
     private $caseConverter;
 
-    public function __construct()
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var string[]
+     */
+    private $fieldNameOverrides;
+
+    public function __construct(LoggerInterface $logger, array $fieldNameOverrides)
     {
         $this->caseConverter = new CamelCaseToSnakeCaseNameConverter(null, false);
+        $this->logger = $logger;
```

Comment:
Allow logger to be null and initialize it to `NullLogger` if done so. This will make the eventual testing easier.

```suggestion
    public function __construct(array $fieldNameOverrides, ?LoggerInterface $logger = null)
    {
        $this->caseConverter = new CamelCaseToSnakeCaseNameConverter(null, false);
        $this->logger = $logger ?? new NullLogger();
```
Don't forget to import the class if this suggestion is acceptable for you.